### PR TITLE
WFCORE-1245 Always report the root cause of deployment failure

### PR DESCRIPTION
**Pull request by spyrkob**
https://issues.jboss.org/browse/WFCORE-1245

### CODE REVIEW

**Comment by spyrkob** [comment]
retest this please

**Comment by bstansberry** [comment]
I put a Hold on this pending some discussion in the JIRA comments.

**Comment by spyrkob** [comment]
@bstansberry I changed the code as we discussed in the JIRA. If it looks OK I'll squash and rebase the PR.

btw I got a bit carried away with refactoring ServiceVerificationHelper and ended up with https://github.com/spyrkob/wildfly-core/commit/0eae8f6a90d93035591f956cc4b2e70116c6a657 - I think it's more readable, but since it's not nessesary for this change, I wasn't sure if it should be bundled with this PR.

File: controller/src/main/java/org/jboss/as/controller/ServiceVerificationHelper.java [hunk 1]
```diff
@@ -168,23 +190,21 @@ private static ModelNode getServiceFailureDescription(final StartException excep
...
         }
 
-        // Only report the missing services if the list includes ones not already tracked
-        // as failed or directly missing
-        Set<ServiceName> retain = new HashSet<ServiceName>(result);
+        Set<ServiceName> retain = new HashSet<>(allMissingServices);
         retain.removeAll(alreadyTracked);
-        if (retain.size() == 0) {
-            result.clear();
+        if (retain.size() != 0) {
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/ServiceVerificationHelper.java]: Shouldn't this be == ?

File: controller/src/main/java/org/jboss/as/controller/ServiceVerificationHelper.java [hunk 2]
```diff
@@ -114,31 +115,36 @@ public synchronized void execute(final OperationContext context, final ModelNode
...
+
+            // print out list of services depending on missing services
+            reportImmediateDependants(problemList, failureDescription);
+
             if (missingTransitive != null) {
                 // See if any other services are known to the service container as being missing and aren't already
                 // tracked by this SVH as failed or directly missing. If any are found, that is some additional
                 // info to the user, so report that
-                SortedSet<ServiceName> allMissing = findAllMissingServices(missingTransitive, trackedServices);
-                if (!allMissing.isEmpty()) {
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/ServiceVerificationHelper.java]: Dropping this means we always report this.

**Comment by spyrkob** [comment]
Sorry, I think I must have changed that for some refactoring I did later and discarded afterwards

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**